[Douglch] iP

Douglch opened the pull request:
# Lele

Lele's an obedient dog bot who'll abide by his user's commands to his best capabilities, including: 
- Keeping track of your tasks
- Checking the status of your tasks
- Last but not least, greeting you with a smile :smiling_face_with_three_hearts:

Here's how to chat with Lele:
1. Download the file [here](https://github.com/Douglch/ip) 
2. Start adding your tasks.

Features:
- [x] Handling Todo tasks
- [ ] Giving a useful description when adding a deadline or event fails

You can change the path directly in `main` method of Lele.java if you have data in another folder:

```java
    public static void main(String[] args) {
        new Lele("./data/lele.txt").run();
    }
```

<sub>Thanks for reviewing my code :) </sub>

Commit:
Add Gradle support

Douglch committed:
meeting level-1 requirements

Douglch committed:
Added break to Level-1 and filled requirements for Level-2

Douglch committed:
Added task class with status marking

Douglch committed:
Added task class

Douglch committed:
fixed task string output

Douglch committed:
Done with Level-3 requirements

Douglch committed:
added todo

Douglch committed:
done with deadline

Douglch committed:
Added Event

Douglch committed:
Added exceptions

Douglch committed:
Added delete functionality

Douglch committed:
Fixed A-TextUiTesting

Douglch committed:
Modified test cases for auto test

Douglch committed:
reformatted code, added TaskList class

Douglch committed:
File add operations added

Douglch committed:
File write operations added

Douglch committed:
Added updating and loading file operations on startup

Douglch committed:
Added updating and loading file operations on startup

Douglch committed:
clean up comments

Douglch committed:
Added date parsing

Douglch committed:
Fix merge conflict

Douglch committed:
Fixed Level-7 not showing correct status on load and finished level-8

Douglch committed:
Merge branch 'add-gradle-support'

Douglch committed:
Added skeleton files for further OOP

Douglch committed:
Add skeleton files for more OOP

Douglch committed:
Merge branch 'master' of github.com:Douglch/ip

Douglch committed:
Refactor accordingly to skeleton files

Douglch committed:
Refactored into packages and clean up main code

Douglch committed:
More OOP with stretch goal

Douglch committed:
Remove useless imports and text adjustment

Douglch committed:
Fix update/load errors, print messages bug

Douglch committed:
Added unit tests

Douglch committed:
Create Jar file

Douglch committed:
Wrote JavaDoc for Duke, UI, tasks

Douglch committed:
Add JavaDocs for the rest

Douglch committed:
Upadted coding standards

Douglch committed:
Add find function

Douglch committed:
Merge branches

Douglch committed:
Fix merge conflict of branch-Level-9

FireRadical22 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,21 @@
+ */
+public abstract class Command {
+
+    public abstract void execute(TaskList taskList, Ui ui, Storage storage)
```
Perhaps you can consider throwing DukeExceptions only and have messages in these exceptions be displayed ui.showError?

FireRadical22 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,130 @@
+     */
+    public static Command parse(String fullCommand) throws UnknownCommandException, EmptyDescException {
+        try {
+            String[] inputArr = fullCommand.split(" ");
```
Perhaps you can consider doing split(" ", 2) to split the input into the key command word and its arguments? If you wish to go with this idea, please consider generating substrings of the argument string without the need to use Stringbuilder.

E.g. for deadline, we can generating description of task from argument by doing substring(0, "/by").

FireRadical22 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,149 @@
+//                } else {
+//                    System.out.println("File is not created");
+//                }
+                dir.mkdir(); f.createNewFile();
```
```suggestion
                dir.mkdir(); 
                f.createNewFile();

```

FireRadical22 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,149 @@
+package duke.storage;
+
+import duke.exception.EmptyStorageException;
+import duke.task.*;
```
Should the imported classes be listed explicitedly?

FireRadical22 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+     *
+     * @param task Task to be added.
+     */
+    public void addTasks(Task task) {
```
```suggestion
    public void addTask(Task task) {

```

FireRadical22 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+     *
+     * @param inputIndex The location of the task in the array list.
+     */
+    public Task deleteTasks(int inputIndex) {
```
```suggestion
    public Task deleteTask(int inputIndex) {

```

FireRadical22 reviewed (commented):
Overall, the code is neat and easy to read. I like the organisation of different commands and exceptions into different classes. However, there were a few places where documentations could be completed.

Douglch committed:
Changed program name

Douglch committed:
Comment changes

Douglch committed:
Fixed majority of checkstyle errors

Douglch committed:
Tutorial Level 1

Douglch committed:
Creating Level-10 branch

Douglch committed:
Complete tutorial

Douglch committed:
Update runtest.sh

Douglch committed:
Added Lele to JavaFx

Douglch committed:
Merge branch 'branch-A-CheckStyle'

Douglch committed:
Resolve merge conflicts from branch-Level-10

Douglch committed:
Added jar files

The previous jar files are not updated to the master branch. There are
also some minor code quality improvements to be made, like unused
imports.

The jar file is built with meta-information within the resource folder.
The unused imports are fixed by removing the java import
lines in Main.java.

Douglch committed:
Added jar files

Douglch committed:
Merge branch 'master' of github.com:Douglch/ip

Douglch committed:
Add java assertions

There currently no way to display any assumptions for the code.

This should be changed to better illustrate the underlying assumptinos
in the code.

This problem is fixed by adding assertions to classes like storage,
parser and TaskList, which helps to show that some lines of code will
not be reached.

Douglch committed:
Improve code quality

Code in several classes do not meet code quality standards (ie. fn addTasks).

Should be changed for better readability.

As such, unused imports are removed and function names are refactored.

Douglch committed:
Code Quality updates

Douglch committed:
Add github workflows for Github Actions

Douglch committed:
Add gradle permissions

Douglch committed:
Remove checkstyle errors

Douglch committed:
Add bash permissions

Douglch committed:
Fix permissions for bash

Douglch committed:
Removed 'change run test permissions' from yml

Douglch committed:
Change bash file permissions

Douglch committed:
Fix text-ui for github actions

Douglch committed:
Fix text-ui #2

Douglch committed:
Commented tests out

Douglch committed:
Saving changes to remote

Douglch committed:
Completed additional extension

Douglch committed:
Merge pull request #1 from Douglch/branch-C-Undo

Branch-C-Undo

Douglch committed:
JavaDoc update for undo

Douglch committed:
Update code quality

Douglch committed:
Add dialog box border

Douglch committed:
Add pane window resizing

Douglch committed:
Add docs folder

Douglch committed:
Add Exit functionality

Douglch committed:
More adjustments to GUI

Douglch committed:
Update README.md

Douglch committed:
Update README.md

Douglch committed:
Update Ui.png and improved GUI

Douglch committed:
Merge branch 'master' of github.com:Douglch/ip

Douglch committed:
Remove Ui.png

Douglch committed:
Add Ui.png

Douglch committed:
Add UG images for UG

Douglch committed:
Update README.md

Douglch committed:
Changed UG images folder name

Douglch committed:
Fix checkstyle errors

Douglch committed:
Add a link to go back to the top of the page

Douglch committed:
fix delete.png name
